examples/siro_sandbox/sandbox_app.py

```diff
@@ -140,59 +128,64 @@ def __init__(self, args, config, gui_input):
             self.gym_habitat_env, config, args, gui_input, self._step_recorder
         )
 
-        self.gui_agent_ctrl = self.ctrl_helper.get_gui_agent_controller()
-
-        self.cam_transform = None
-        self.cam_zoom_dist = 1.0
-        self._max_zoom_dist = 50.0
-        self._min_zoom_dist = 0.02
-
-        self.gui_input = gui_input
-
-        self._debug_line_render = None
         self._debug_images = args.debug_images
 
-        self._viz_anim_fraction = 0.0
-
-        self.lookat = None
-
-        if self.is_free_camera_mode() and args.first_person_mode:
+        is_free_camera_mode = (
+            self.ctrl_helper.get_gui_controlled_agent_index() is None
+        )
+        if is_free_camera_mode and args.first_person_mode:
             raise RuntimeError(
                 "--first-person-mode must be used with --gui-controlled-agent-index"
             )
 
-        # lookat offset yaw (spin left/right) and pitch (up/down)
-        # to enable camera rotation and pitch control
-        self._first_person_mode = args.first_person_mode
-        if self._first_person_mode:
-            self._lookat_offset_yaw = 0.0
-            self._lookat_offset_pitch = float(
-                mn.Rad(mn.Deg(20.0))
-            )  # look slightly down
-            self._min_lookat_offset_pitch = (
-                -max(min(np.radians(args.max_look_up_angle), np.pi / 2), 0)
-                + 1e-5
+        if is_free_camera_mode:
+            raise RuntimeError(
+                "Free camera mode is temporarily unsupported! you must set --gui-controlled-agent-index."
             )
-            self._max_lookat_offset_pitch = (
-                -min(max(np.radians(args.min_look_down_angle), -np.pi / 2), 0)
-                - 1e-5
+
+        self._viz_anim_fraction = 0.0
+        self._pending_cursor_style = None
+
+        def local_end_episode(do_reset=False):
+            self._end_episode(do_reset)
+
+        self._sandbox_service = SandboxService(
```

Comment:
This is roughly the "DI container" software pattern.

Reply:
My understanding is that SandboxService is the interface that should be used by (passed to) app state objects only. 

We shouldn't keep it as an instance attribute and if we need to access attributes / call methods from within  SandboxDriver we should call them directly.